proto/rill/runtime/v1/catalog.proto

```diff
@@ -53,6 +66,8 @@ message Model {
   Dialect dialect = 3;
   // Detected schema of the model
   StructType schema = 4;
+  // To materialize model or not
+  Materialize materialize = 5;
```

Comment:
I also thought about it but end up keeping the detailed info in some case we need to store in catalog in future then we will have backwards compatibility issue. Another case is if we want to do something extra in case of switching from true to inferred or false to unspecified and vice versa and the model file is edited offline, in this case we will loose previous value for materialize if not stored in catalog. Both are my apprehensions not sure how useful.

Reply:
I would vote for keeping it simple for now with just a `bool`, and then we can augment with parsing data later if needed

Reply:
changed. After the change the invalid materialize error is returned as `ReconcileError_CODE_SYNTAX` not as `ReconcileError_CODE_VALIDATION` as we get this error during artifact read and not model validation.  If we want this then that will require creating custom error struct and checking for this type. We also don't perform sql validation.